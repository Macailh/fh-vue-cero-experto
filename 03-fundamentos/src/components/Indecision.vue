<template>
  <img v:if="img" :src="img" alt="bg">
  <h2>Indecision</h2>

  <div class="indecision-container">
    <input type="text" placeholder="Hazme una pregunta" v-model="question">
    <p>Recuerda terminal con un signo de interrogaci√≥n (?)</p>

    <div v-if="isValidQuestion">
        <h2>{{question}}</h2>
        <h1>{{answer === 'yes' ? 'Si!!!' : 'No!!!'}}</h1>
    </div>
  </div>
</template>

<script>
export default {
    data() {
        return {
            question: null,
            answer: null,
            img: null,
            isValidQuestion: false
        }
    },
    methods: {
      async getAnwer() {
        this.answer = 'Pensando...'

        const {answer, image} = await fetch('https://yesno.wtf/api').then(r => r.json())
        this.answer = answer
        this.img = image
      }
    },
    watch: {
      question(value, oldValue) {
        // console.log({value, oldValue})
        // console.log(value.includes('?'))
        this.isValidQuestion = false
        if(!value.includes('?')) return
        this.isValidQuestion = true
        this.getAnwer()
      }
    }

}
</script>

<style scoped>
  img, .bg-dark {
    height: 100vh;
    left: 0px;
    max-height: 100%;
    min-width: 100%;
    position: fixed;
    top: 0px;
    width: 100vw;
  }

  .bg-dark {
    background-color: rgba(0, 0, 0, 0.4);
  }

  .indecision-container {
    position: relative;
    z-index: 99;
  }

  input {
    width: 250px;
    padding: 10px 15px;
    border-radius: 5px;
    border: none;
  }

  input:focus {
    outline: none;
  }

  p {
    color: white;
    font-size: 20px;
    margin-top: 3px;
  }

  h1, h2 {
    color: white;
  }

  h2 {
    margin-top: 150px;
  }
</style>