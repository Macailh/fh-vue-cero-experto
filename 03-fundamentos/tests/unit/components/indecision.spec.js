import { shallowMount } from "@vue/test-utils";
import Indecision from "@/components/Indecision.vue";


describe('Indecision component', () => {
    let wrapper
    let clgSpy

    beforeEach(() => {
        wrapper = shallowMount(Indecision)
        clgSpy = jest.spyOn(console, 'log')
    })

    test('snapshot match', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('writing to the input should not write anything (console.log)', async() => {
        const getAnswerSpy = jest.spyOn(wrapper.vm, 'getAnswer')
        const input = wrapper.find('input')
        await input.setValue('test text')

        expect(clgSpy).toHaveBeenCalled()
        expect(getAnswerSpy).not.toHaveBeenCalled()
    })

    test('when writing the symbol of "?" should fire the fetch', () => {

    })

    test("tests in getAnswer() method", () => {

    })

    test("test in getAnswer() method - API fail", () => {

    })
})