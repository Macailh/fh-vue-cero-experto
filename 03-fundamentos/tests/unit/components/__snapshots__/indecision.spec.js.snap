// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indecision component snapshot match 1`] = `
<img v:if="img" alt="bg">
<h2>Indecision</h2>
<div class="indecision-container"><input type="text" placeholder="Hazme una pregunta">
  <p>Recuerda terminar con un signo de interrogaci√≥n (?)</p>
  <!--v-if-->
</div>
`;
